pkg/api/team_members.go

```diff
@@ -152,10 +154,20 @@ func (hs *HTTPServer) RemoveTeamMember(c *models.ReqContext) response.Response {
 	return response.Success("Team Member removed")
 }
 
-// addTeamMember adds a team member.
+// addOrUpdateTeamMember adds or updates a team member.
 //
 // Stubbable by tests.
-var addTeamMember = func(sqlStore *sqlstore.SQLStore, userID, orgID, teamID int64, isExternal bool,
+var addOrUpdateTeamMember = func(resourcePermissionService *resourcepermissions.Service, userID, orgID, teamID int64, isExternal bool,
```

Comment:
If we pass the permission string instead of the model we can reuse this part for adding, updating and removing